packages/react-meteor-data/useTracker.js

```diff
@@ -0,0 +1,155 @@
+/* global Meteor, Package, Tracker */
+import React, { useState, useEffect, useRef } from 'react';
+
+// Use React.warn() if available (should ship in React 16.9).
+const warn = React.warn || console.warn.bind(console);
+
+// Warns if data is a Mongo.Cursor or a POJO containing a Mongo.Cursor.
+function checkCursor(data) {
+  let shouldWarn = false;
+  if (Package.mongo && Package.mongo.Mongo && data && typeof data === 'object') {
+    if (data instanceof Package.mongo.Mongo.Cursor) {
+      shouldWarn = true;
+    } else if (Object.getPrototypeOf(data) === Object.prototype) {
+      Object.keys(data).forEach((key) => {
+        if (data[key] instanceof Package.mongo.Mongo.Cursor) {
+          shouldWarn = true;
+        }
+      });
+    }
+  }
+  if (shouldWarn) {
+    warn(
+      'Warning: your reactive function is returning a Mongo cursor. '
+      + 'This value will not be reactive. You probably want to call '
+      + '`.fetch()` on the cursor before returning it.'
+    );
+  }
+}
+
+// taken from https://github.com/facebook/react/blob/
+// 34ce57ae751e0952fd12ab532a3e5694445897ea/packages/shared/objectIs.js
+function is(x, y) {
+  return (
+    (x === y && (x !== 0 || 1 / x === 1 / y))
+    || (x !== x && y !== y) // eslint-disable-line no-self-compare
+  );
+}
+
+// inspired by https://github.com/facebook/react/blob/
+// 34ce57ae751e0952fd12ab532a3e5694445897ea/packages/
+// react-reconciler/src/ReactFiberHooks.js#L307-L354
+// used to replicate dep change behavior and stay consistent
+// with React.useEffect()
+function areHookInputsEqual(nextDeps, prevDeps) {
+  if (prevDeps === null || prevDeps === undefined || !Array.isArray(prevDeps)) {
```

Comment:
I initially did this because of https://github.com/facebook/react/blob/34ce57ae751e0952fd12ab532a3e5694445897ea/packages/react-reconciler/src/ReactFiberHooks.js#L318 and https://github.com/facebook/react/blob/34ce57ae751e0952fd12ab532a3e5694445897ea/packages/react-reconciler/src/ReactFiberHooks.js#L878

Which only checks for `null` and `undefined`. Afaik `if (!prevDeps || !Array.isArray(prevDeps) {` should work as well since `0` is not expected anyway, also might be easier to think, I agree with that change.

Reply:
I was thinking that a "falsy" value like null or undefined is actually valid. This check though will kick up a warning if null or undefined are used.

Reply:
Oh I see that React doesn't allow the value to change. This is a mistake IMHO, because React will sometimes reuse components in ways that are not always clear (in your react-router `Switch` case for example). I think we should assume those values can change at any time (and our component does a great job of allowing that - with the possible exception of the `useEffect` hooks, which takes unfiltered `deps` - more edge case shenanigans from upstream...)